### PR TITLE
VACMS-13848: Creates strategyresolver service.

### LINKED ISSUE #13848: Create new `ContentReleaseStrategyResolver` service.
## Description

The `ContentReleaseStrategyResolver` encapsulates the logic to translate the perceived environment, as retrieved from `EnvironmentDiscovery` (see #13845), into an appropriate content release strategy plugin (see #13847).

The strategies themselves are managed as plugins. This service's primary method vends a strategy plugin from its injected `ContentReleaseStrategyPluginManager` service according to the environment indicator provided by its injected `EnvironmentDiscovery` service.

This should allow greater flexibility during the development process, as currently local content releases from non-default branches have to be dispatched from the `content-build` project via command line. Also, this can be useful during the Accelerated Publishing development process, as it should allow incremental changes without necessitating long-running feature branches within the CMS repo.

## Acceptance Criteria

- [x] A `ContentReleaseStrategyResolverInterface` interface exists with the following methods:
  - [x] `getStrategy(): ContentReleaseStrategyPluginInterface`
  - [x] `triggerContentRelease(): void`
- [x] A `ContentReleaseStrategyResolver` class exists, implementing `ContentReleaseStrategyResolverInterface`.
  - [x] Unit tests exist for all functionality.

### PR DESCRIPTION
**Pull request by ndouglas**
Closes #13848.

Also includes some random refactoring that came up in the course of doing this work and adding tests. I'll note that work in separate comments.

### CODE REVIEW

File: docroot/modules/custom/va_gov_content_release/va_gov_content_release.services.yml [hunk 1]
```diff
@@ -1,10 +1,16 @@
 services:
   va_gov_content_release.github_repository_dispatch:
-    class: Drupal\va_gov_content_release\GitHubRepositoryDispatch
+    class: Drupal\va_gov_content_release\GitHub\GitHubRepositoryDispatch
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_content_release/va_gov_content_release.services.yml:3]: (Oops)

File: docroot/modules/custom/va_gov_content_release/src/LocalFilesystem/LocalFilesystemBuildFile.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_content_release/src/LocalFilesystem/LocalFilesystemBuildFile.php:1]: Split out the critical code into a separate service for testability.

File: tests/phpunit/va_gov_content_release/functional/Reporter/ReporterTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_content_release/functional/Reporter/ReporterTest.php:1]: Functional test to verify the service can be constructed.

File: tests/phpunit/va_gov_content_release/functional/Strategy/Resolver/ResolverTest.php
Comment (ndouglas) [inline tests/phpunit/va_gov_content_release/functional/Strategy/Resolver/ResolverTest.php:1]: Functional test to verify the service can be constructed.

**Review by edmund-dunn — APPROVED** [review]
Wicked good code!

**Pull request merged**